### PR TITLE
(foreman role) add foreman discovery plugin

### PR DESCRIPTION
**Pull request by jhoblitt**
Related to https://github.com/lsst-it/foreman_templates/pull/2

### CODE REVIEW

File: hieradata/role/foreman.yaml [hunk 1]
```diff
@@ -106,6 +106,9 @@ foreman_proxy::log_buffer: 2000
 foreman_proxy::log_buffer_errors: 1000
 foreman_proxy::log_level: "DEBUG"
 foreman_proxy::log: "/var/log/foreman-proxy/proxy.log"
+foreman_proxy::plugin::discovery::image_name: "fdi-image-4.99.99-6224850.tar"
+foreman_proxy::plugin::discovery::install_images: true
+foreman_proxy::plugin::discovery::source_url: "https://github.com/lsst-it/foreman-discovery-image/releases/download/lsst-4.99.99/"
```
[the remark this answers is not in the document]
Comment (jhoblitt) [reply hieradata/role/foreman.yaml:111]: The `foreman_proxy::plugin::discovery` class is https://github.com/theforeman/puppet-foreman_proxy/blob/master/manifests/plugin/discovery.pp
The firewalld/nft tftp fix is https://github.com/theforeman/foreman-discovery-image/pull/151

[the remark this answers is not in the document]
Comment (jhoblitt) [reply hieradata/role/foreman.yaml:111]: The fact which is being put in `udev_fact.zip` I am proposing to add to the `puppet/systemd` module. https://github.com/voxpupuli/puppet-systemd/pull/292

**Comment by jhoblitt** [comment]
I'm going to change this PR back to draft status to look at BMC management.  When discovery creates a host, it is add a BMC type interface to the host, which is sort of strange as BMC support is disabled.

**Comment by jhoblitt** [comment]
I have enabled foreman's bmc support (and manually tested it as working with one of the new k8s nodes) and de-duplicated some of the tests.

File: hieradata/role/foreman.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbarria) [reply hieradata/role/foreman.yaml:113]: thanks! i follow up those to get the entire picture, this was just great work (https://github.com/theforeman/foreman-discovery-image/pull/151)

**Review by cbarria — APPROVED** [review]
great work!

**Pull request merged**